diff --git a/node_modules/@utexo/rgb-sdk-rn/ios/Rgb.mm b/node_modules/@utexo/rgb-sdk-rn/ios/Rgb.mm
index 3d2fe34..c287269 100644
--- a/node_modules/@utexo/rgb-sdk-rn/ios/Rgb.mm
+++ b/node_modules/@utexo/rgb-sdk-rn/ios/Rgb.mm
@@ -128,7 +128,7 @@ - (void)rlnUnlockNode:(double)nodeId
   bitcoindRpcUsername:(NSString * _Nullable)bitcoindRpcUsername
   bitcoindRpcPassword:(NSString * _Nullable)bitcoindRpcPassword
       bitcoindRpcHost:(NSString * _Nullable)bitcoindRpcHost
-      bitcoindRpcPort:(NSNumber * _Nullable)bitcoindRpcPort
+      bitcoindRpcPort:(NSNumber *)bitcoindRpcPort
             indexerUrl:(NSString *)indexerUrl
          proxyEndpoint:(NSString *)proxyEndpoint
      announceAddresses:(NSArray<NSString *> *)announceAddresses
@@ -939,7 +939,7 @@ - (void)rlnUnlockNodeWithNativeExternalSigner:(double)nodeId
                           bitcoindRpcUsername:(NSString * _Nullable)bitcoindRpcUsername
                           bitcoindRpcPassword:(NSString * _Nullable)bitcoindRpcPassword
                               bitcoindRpcHost:(NSString * _Nullable)bitcoindRpcHost
-                              bitcoindRpcPort:(NSNumber * _Nullable)bitcoindRpcPort
+                              bitcoindRpcPort:(NSNumber *)bitcoindRpcPort
                                    indexerUrl:(NSString *)indexerUrl
                                 proxyEndpoint:(NSString *)proxyEndpoint
                             announceAddresses:(NSArray<NSString *> *)announceAddresses
